include/GameLogic.h

```diff
@@ -27,16 +28,14 @@ class GameLogic {
         Enemy enemy4;
         Platform floor;
         Platform platformA;
-        Platform platformB;
-        Platform platformC;
-        Platform platformD;
-        Platform platformE;
-        Projectile projectile;
-        Spike spike1;
-        Spike spike2;
-        Spike spike3;
-        Spike spike4;
+
+        LevelCreator loader;
+
         std::vector<std::reference_wrapper<Actor>> actorsVector;
+        std::vector<std::vector<Actor>> nestedActors;
+        std::vector<Platform> platVector;
+        std::vector<Spike> spikeVector;
+        std::vector<Enemy> enemyVector;
```

Comment:
What is the purpose of all these seemingly duplicate vectors? Also, rename platVector to platformVector.

Reply:
Yeah it looks like they are duplicates of the following vectors, however these are different because they don't use reference wrappers. @carter-greis knows better than I do about the difference in functionality, but I know that when loading in the object from level loader they cannot go directly into the reference wrapper.

Reply:
So I tried defining them just locally in the setLevel method, but then the objects weren't staying defined after the method ended and it wasn't working. I still think the conversion to reference_wrapper is needed because otherwise the objects stored in the actorsVector and the individual vectors for the specific types would not be the same object and thus changes made from the specific vectors would not affect the appropriate object in the actorsVector. I believe that is why I originally made them reference_wrappers. I am travelling today and won't have access to my computer until tonight, but I will try to take a look at it and make sure that my reasoning is still valid.